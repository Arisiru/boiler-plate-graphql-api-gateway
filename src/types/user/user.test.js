const schema = require('./schema');
const resolver = require('./resolver');
const mockUsers = require('../../mock/users');

describe('[Type.User]', () => {
  it('should compile schema', async () => {
    expect(schema).toMatchSnapshot();
  });

  const user = Object.freeze(mockUsers[0]);

  it('maps user to "firstName" property', () => {
    expect(resolver.User.firstName(user)).toEqual(user.firstName);
  });

  it('maps user to "lastName" property', () => {
    expect(resolver.User.lastName(user)).toEqual(user.lastName);
  });

  it('maps user to "email" property', () => {
    expect(resolver.User.email(user)).toEqual(user.email);
  });

  it('maps user to "role" property', () => {
    expect(resolver.User.role(user)).toEqual(user.role);
  });

  it('maps user to "photo" property', () => {
    expect(resolver.User.photo(user)).toEqual(user.photo);
  });
});
