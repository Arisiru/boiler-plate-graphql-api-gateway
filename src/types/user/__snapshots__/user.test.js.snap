// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Type.User] should compile schema 1`] = `
Object {
  "definitions": Array [
    Object {
      "description": undefined,
      "directives": Array [],
      "fields": Array [
        Object {
          "arguments": Array [],
          "description": undefined,
          "directives": Array [],
          "kind": "FieldDefinition",
          "name": Object {
            "kind": "Name",
            "value": "email",
          },
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
        },
        Object {
          "arguments": Array [],
          "description": undefined,
          "directives": Array [],
          "kind": "FieldDefinition",
          "name": Object {
            "kind": "Name",
            "value": "firstName",
          },
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
        },
        Object {
          "arguments": Array [],
          "description": undefined,
          "directives": Array [],
          "kind": "FieldDefinition",
          "name": Object {
            "kind": "Name",
            "value": "lastName",
          },
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
        },
        Object {
          "arguments": Array [],
          "description": undefined,
          "directives": Array [],
          "kind": "FieldDefinition",
          "name": Object {
            "kind": "Name",
            "value": "role",
          },
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "String",
            },
          },
        },
        Object {
          "arguments": Array [],
          "description": undefined,
          "directives": Array [],
          "kind": "FieldDefinition",
          "name": Object {
            "kind": "Name",
            "value": "photo",
          },
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "String",
            },
          },
        },
      ],
      "interfaces": Array [],
      "kind": "ObjectTypeDefinition",
      "name": Object {
        "kind": "Name",
        "value": "User",
      },
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 118,
    "start": 0,
  },
}
`;
